se/sics/mspsim/ui/ControlUI.java

```diff
@@ -97,18 +98,37 @@ private void setup() {
     jp.add(this, BorderLayout.WEST);
     jp.add(dui = new DebugUI(cpu), BorderLayout.CENTER);
     window.add(jp);
-    
+
     createButton("Debug On");
     controlButton = createButton(cpu.isRunning() ? "Stop" : "Run");
-    stepAction = new AbstractAction("Single Step") {
+
+    JButton stepButton = new JButton();
+    // Allow the button to get the same height as the other buttons.
+    stepButton.setMaximumSize(new Dimension(60, Short.MAX_VALUE));
+
+    SpinnerModel stepsModel = new SpinnerNumberModel();
```

Comment:
Perhaps restrict the model to only positive number of steps?

```suggestion
    SpinnerNumberModel stepsModel = new SpinnerNumberModel();
    stepsModel.setValue(1);
    stepsModel.setMinimum(1);
```

Reply:
Yes, fixed in latest revision.